### PR TITLE
You can now set the max_key_fee value to null with cli.

### PR DESCRIPTION
**Pull request by Nykseli**
Fixes https://github.com/lbryio/lbry-sdk/issues/2050

### CODE REVIEW

**Comment by tzarebczan** [comment]
Hey @Nykseli , congrats on the first contribution! We'll get a PR review on this. Did you try via both CLI and json? 

Can we show you some [appreciation](https://lbry.com/faq/appreciation) for the PR?

**Comment by Nykseli** [comment]
Yea I tried via both CLI and json. Both methods work

File: lbry/lbry/conf.py [hunk 1]
```diff
@@ -170,6 +170,11 @@ def deserialize(self, value):
                 'amount': float(value['amount']),
             }
         if isinstance(value, str):
+            if not value:
+                return None
```
Comment (eukreign) [inline lbry/lbry/conf.py]: Can you combine this into the initial `if value is None:` and just make that one `if value:`.

File: lbry/lbry/conf.py [hunk 2]
```diff
@@ -170,6 +170,11 @@ def deserialize(self, value):
...
+            if value.lower() == "null":
+                return None
+
```
Comment (eukreign) [inline lbry/lbry/conf.py]: Don't need the extra blank line here.

File: lbry/lbry/conf.py (hunk 1 above)
Comment (Nykseli) [reply lbry/lbry/conf.py]: Yes. That works and is a lot cleaner

File: lbry/lbry/conf.py (hunk 2 above)
Comment (Nykseli) [reply lbry/lbry/conf.py]: I'll clean that up

**Comment by Nykseli** [comment]
I can add a unit test for the json to [tests_conf.py](https://github.com/lbryio/lbry-sdk/blob/master/lbry/tests/unit/test_conf.py). Should I also add a test for the cli? I believe that should go to [test_cli.py](https://github.com/lbryio/lbry-sdk/blob/master/lbry/tests/unit/test_cli.py)

**Comment by eukreign** [comment]
> I can add a unit test for the json to [tests_conf.py](https://github.com/lbryio/lbry-sdk/blob/master/lbry/tests/unit/test_conf.py). Should I also add a test for the cli? I believe that should go to [test_cli.py](https://github.com/lbryio/lbry-sdk/blob/master/lbry/tests/unit/test_cli.py)

Just in [tests_conf.py](https://github.com/lbryio/lbry-sdk/blob/master/lbry/tests/unit/test_conf.py) is fine.

**Comment by Nykseli** [comment]
Its seems you need to have the daemon running if you want to properly test the json and cli since they both use the Daemon.jsonrpc_settings_set method. The method basically just calls the deserialize method of the given Setting and updates the config. So we could just test the deserialize method like this:

```
def test_max_key_fee_from_json(self):
        mkf = MaxKeyFee('The test', {'currency': 'USD', 'amount': 50.0})
        c = Config.create_from_arguments([])

        d = mkf.deserialize('null')
        setattr(c, 'max_key_fee', d)
        self.assertEqual(c.max_key_fee, None)
        d = mkf.deserialize('1.0 BTC')
        setattr(c, 'max_key_fee', d)
        self.assertEqual(c.max_key_fee, {'amount': 1.0, 'currency': 'BTC'})
        d = mkf.deserialize('')
        setattr(c, 'max_key_fee', d)
        self.assertEqual(c.max_key_fee, None)
```

Is this testing method enough for now? It would be easier to add proper test to the test_Daemon.py but the tests are skipped for some reason

**Comment by eukreign** [comment]
@Nykseli I think if you can just update the tests mentioned in my previous comment it would be enough. https://github.com/lbryio/lbry-sdk/pull/2292#pullrequestreview-258605261

**Comment by Nykseli** [comment]
Alright. The test is updated

**Comment by eukreign** [comment]
@Nykseli the test does not appear to test your changes.

**Comment by tzarebczan** [comment]
@Nykseli did you want to take a final shot at this? Thanks again for your effort and patience. Once this is merged, can we show you some [appreciation](https://lbry.com/faq/appreciation) for the PR?

**Comment by eukreign** [comment]
closing due to inactivity, we can re-open this later if necessary

**Pull request closed without merging**